Sources/MusicXML/InternalTypes/AttributeGroup.swift

```diff
@@ -5,4 +5,5 @@
 //  Created by Benjamin Wetherfield on 11/18/19.
 //
 
-internal protocol AttributeGroup {}
+import XMLCoder
+internal protocol AttributeGroup: DynamicNodeEncoding {}
```

Comment:
Can a default implementation of `nodeEncoding(for:)` be placed here?

Also, have you checked that the `nodeEncoding(for:)` is necessary at the `AttributeGroup` level? When I was messing with things, it only seemed to matter that the attribute-group-containing structure implemented `DynamicNodeEncoding/Decoding` as such. I could be very wrong, though.

Reply:
Are you sure it is possible to implement functions in protocols? I think that was the issue (I'd like to but I don't know if it's possible)

Reply:
Doesn't make sense to me that this would work for encoding within the attribute-group-containing structure as the relevant `CodingKeys` don't actually have to be in there, if that makes sense?

Reply:
Of course, it would work either way for decoding. I think the problems emerge with encoding.